File: src/main/java/com/menighin/luwak/core/interfaces/ILuwakDatasource.java
package com.menighin.luwak.core.interfaces;

import com.menighin.luwak.core.dtos.CrudResponse;

import javax.validation.constraints.NotNull;
import java.util.ArrayList;

public interface ILuwakDatasource<M extends ILuwakModel, F extends ILuwakFilter> {

	M getById(int id);

	ArrayList<M> getAll(Integer masterId, Integer page, F filter);

	boolean create(Integer masterId, ILuwakDto luwakDto);

	boolean update(Integer masterId, int id, ILuwakDto luwakDto);

	boolean delete(Integer masterId, int id);

	boolean deleteMany(Integer masterId, @NotNull int[] ids);

	Integer count(Integer masterId);
}

File: src/main/java/com/menighin/luwak/core/interfaces/ILuwakDatasource.kt
package com.menighin.luwak.core.interfaces

interface ILuwakDatasource<M : ILuwakModel, F : ILuwakFilter> {

	fun getById(id: Long): M

	fun getAll(masterId: Long?, page: Int?, filter: F?, sort: HashMap<String, String>?): List<M>

	fun create(masterId: Long?, luwakDto: ILuwakDto): Boolean

	fun update(masterId: Long?, id: Long, luwakDto: ILuwakDto): Boolean

	fun delete(masterId: Long?, id: Long): Boolean

	fun deleteMany(masterId: Long?, ids: LongArray): Boolean

	fun count(masterId: Long?, filter: F?): Int
}
